src/tech/v3/protocols/column.clj

```diff
@@ -45,3 +45,19 @@ Supported types are:
     "Convert to a java primitive array of a given datatype.  For strings,
 an implicit string->double mapping is expected.  For booleans, true=1 false=0.
 Finally, any missing values should be indicated by a NaN of the expected type."))
+
+
+(defprotocol PHasIndexStructure
+  (index-structure [this]
+    "Returns an index-structure for the column.")
+  (with-index-structure
+    [this datatype-keyword klass custom-make-index-structure-fn]
+    "Returns a copy of the column that will return an index-structure using the
+provided `custom-make-index-strucutre-fn`."))
+
```

Comment:
As noted before I think we should have have datatype-keyword and klass here.  Those don't always apply in the average case is that I have a particular dataset and I have a particular indexing pathway I want to apply but the datatype in the column is a long or double or something like that.